src/pages/workspaces/WorkspacesList/WorkspaceUserActions.ts

```diff
@@ -0,0 +1,24 @@
+import { createContext } from 'react';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+export const WorkspaceUserActionsContext = createContext<WorkspaceUserActionsState>({
+  userActions: {
+    creatingNewWorkspace: false,
+  },
+  setUserActions: () => {},
+});
+
+export interface WorkspaceUserActionsState {
+  userActions: WorkspaceUserActions;
+  setUserActions: (actions: Partial<WorkspaceUserActions>) => void;
+}
+
+export interface WorkspaceUserActions {
+  creatingNewWorkspace: boolean;
```

Comment:
All of the other actions are operating on a workspace. This seems like it doesn't belong here.

Reply:
They're actions the user can take on the workspace list page that needs to be tracked for modals.

creatingNewWorkspace is a boolean because if you're creating a new workspace, it doesn't have an id yet.

I expect that once Miguel's modal manager component lands, this will all be switched to that.

Reply:
Sounds good. Wasn't sure if the plan was to still use Context instead of props plumbing even with the modal manager.